### DISCUSSION TITLE
NextJS 14 with **pages** routing and MSW 2.x

### DISCUSSION THREAD

**Discussion by FredLavoie** [mswjs/msw#2007] [Q&A] [answered]
I'm trying to upgrade MSW to v2.1.2 in my NextJS project which only uses the pages routing (not app routing), and I have gotten the individual components to pass the test with API calls being mocked by MSW, but I can't get my pages which have 1-2 API calls each to run. I'm not making any API calls within `getServerSideProps`.

I get the `ReferenceError: Response is not defined`. I understand that the migration guide proposes a solution for this, which includes setting `testEnvironmentOptions: { customExportConditions: [''] },` in the `jest.config.ts` file. But now I'm getting this error:

```
The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string. 
Consider using the "jsdom" test environment.

ReferenceError: document is not defined
```

I can post some of my configs. But my question is, can MSW work with the pages components that make API calls?

**Comment by kettanaito** [comment, marked as answer]
Please follow this pull request for updates on MSW + Next.js 14: https://github.com/mswjs/examples/pull/101. Thanks.
